packages/bpk-component-graphic-promo/src/BpkGraphicPromo-test.js

```diff
@@ -0,0 +1,130 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow strict */
+
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import { cssModules } from 'bpk-react-utils';
+
+import BpkGraphicPromo from './BpkGraphicPromo';
+import TEXT_ALIGN from './BpkGraphicPromoTextAlign';
+import STYLES from './BpkGraphicPromo.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+const props = {
+  kicker: 'Kicker',
+  headline: 'Ride your wave',
+  strapline:
+    'Portugal and 6 more countries have just been added to the UK travel green list',
+  image: 'path/to/image.jpg',
+  sponsor: {
+    label: 'Sponsored',
+    logo: 'path/to/logo.png',
+    altText: 'Airline Name',
+  },
+  ctaText: 'Learn more',
+  onClick: jest.fn(),
+  textAlign: TEXT_ALIGN.start,
+};
+
+describe('BpkGraphicPromo', () => {
+  beforeEach(() => {
+    props.onClick.mockReset();
+  });
+
+  it('should render correctly with all properties set', () => {
+    const { asFragment } = render(<BpkGraphicPromo {...props} />);
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should support custom class names', () => {
+    const { asFragment } = render(
+      <BpkGraphicPromo className="custom-classname" {...props} />,
+    );
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should support arbitrary props', () => {
+    const { asFragment } = render(<BpkGraphicPromo testid="123" {...props} />);
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should render correctly when centre aligned', () => {
+    const customProps = { ...props, textAlign: 'center' };
+    const { asFragment } = render(<BpkGraphicPromo {...customProps} />);
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should render correctly when right aligned', () => {
+    const customProps = { ...props, textAlign: 'end' };
+    const { asFragment } = render(<BpkGraphicPromo {...customProps} />);
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should render correctly when inverted portrait', () => {
+    const customProps = { ...props, invertVertically: true };
+    const { asFragment } = render(<BpkGraphicPromo {...customProps} />);
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should not display kicker or strapline when not provided', () => {
+    const customProps = { ...props };
+    delete customProps.kicker;
+    delete customProps.strapline;
+    const { asFragment } = render(<BpkGraphicPromo {...customProps} />);
+
+    expect(asFragment()).toMatchSnapshot();
```

Comment:
Can we avoid snapshot tests when we have a specific assertion, such as not displaying elements?

`not.toBeInTheDocument()` etc gives a lot stronger assertions

Reply:
Agreed. I prefer checking for the elements too. The rest of Backpack is using snapshots though. 😅